src/components/Tabs/Annotation/AnnotationFileInfo.tsx

```diff
@@ -0,0 +1,130 @@
+import { Card } from "antd";
+import React, { ReactElement, useContext, useMemo } from "react";
+
+import { FlexColumn, FlexRow } from "../../../styles/utils";
+import { renderStringArrayAsJsx } from "../../../utils/formatting";
+
+import { AnnotationParseResult } from "../../../colorizer/AnnotationData";
+import { AppThemeContext } from "../../AppStyle";
+import IconButton from "../../IconButton";
+import MessageCard from "../../MessageCard";
+
+type AnnotationFileInfoProps = {
+  errorText: string;
+  file: File | null;
+  parseResult: AnnotationParseResult | null;
+  clearFile: () => void;
+};
+
+function formatQuantityString(quantity: number, singular: string, plural: string): string {
+  return `${quantity} ${quantity === 1 ? singular : plural}`;
+}
+
+/**
+ * Displays information about a parsed annotation file.
+ * Shows the file name, number of objects parsed, labels, and any warnings about mismatched data.
+ */
+export default function AnnotationFileInfo(props: AnnotationFileInfoProps): ReactElement {
+  const { errorText, file: uploadedFile, parseResult } = props;
+  const theme = useContext(AppThemeContext);
+
+  const conversionWarnings = useMemo((): string[] => {
+    if (!parseResult) {
+      return [];
+    }
+    const conversionWarnings: string[] = [];
+    const { invalidIds, mismatchedTimes, mismatchedTracks, unparseableRows } = parseResult;
+    if (invalidIds >= 1) {
+      const warningText = formatQuantityString(invalidIds, "object had an ID that is", "objects had IDs that are");
+      conversionWarnings.push(`- ${warningText} not in the dataset and could not be parsed.`);
+    }
+    if (unparseableRows >= 1) {
+      const warningText = formatQuantityString(unparseableRows, "object", "objects");
+      conversionWarnings.push(`- ${warningText} had non-numeric values in a metadata column and could not be parsed.`);
+    }
+    const maxMismatchedData = Math.max(mismatchedTimes, mismatchedTracks);
+    if (maxMismatchedData >= 1) {
+      const warningText = formatQuantityString(
+        maxMismatchedData,
+        "object had a time or track that does",
+        "objects had times or tracks that do"
+      );
+      conversionWarnings.push(`- ${warningText} not match the current dataset.`);
+    }
+    return conversionWarnings;
+  }, [parseResult]);
+
+  const getFileInfoContents = (): ReactElement => {
+    const hasError = errorText !== "";
+
+    // Show only error message if there was an error
+    if (hasError || !parseResult) {
+      return (
+        <MessageCard type="error">
+          <p style={{ color: theme.color.text.error }}>{errorText || "File could not be parsed."}</p>
+        </MessageCard>
+      );
+    }
+
+    const parsedObjects = parseResult.totalRows - parseResult.unparseableRows - parseResult.invalidIds;
+    const totalObjects = parseResult.totalRows;
+    let parsedObjectsText = parsedObjects === totalObjects ? parsedObjects : `${parsedObjects}/${totalObjects}`;
```

Comment:
nit: This is working well, but I'm wondering if we could refactor this section to use the helper function `formatQuantityString` instead?

Reply:
Hmm, I'd probably need to make a separate helper method because the formatting on this one is a little different... I'll take a stab at it!